examples/object_detection/ios/ObjectDetector/Views/PreviewView.swift

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import AVFoundation
+
+/**
+ Displays a preview of the image being processed. By default, this uses the device's camera frame,
+ but will use a still image copied from clipboard if `shouldUseClipboardImage` is set to true.
+ */
+class PreviewView: UIView {
```

Comment:
This is only used in the CameraViewController, and the logic is quite simple. Rather than defining a custom class, can you just implement it in `CameraViewController` to reduce the number of classes?

Reply:
Can you add it as a sublayer like this?
https://guides.codepath.com/ios/Creating-a-Custom-Camera-View#step-11-configure-the-live-preview

Reply:
I tried this. The frame of the added sublayer has to be updated programatically in viewWilllayoutsubviews when the device orientation changes. There is a slight glitch as the rotation and subsequent updation of the frame. I tried making this more smooth. But don't think it is possible.

https://github.com/googlesamples/mediapipe/assets/42828735/8c0d71d6-bc90-417f-b1a5-58792842b3a7

As opposed to overriding the layer subclass in a custom view subclass (Preview View). UIKit handles the rotation along with the view in this case. We don't need to update the frame anywhere.

https://github.com/googlesamples/mediapipe/assets/42828735/04bad040-743c-4dde-893d-c2f102d825a9

If this glitch during rotation and the extra frame updates are okay to include, I'll remove the PreviewView subclass and make it a sublayer. LMK

Reply:
I think the glitch is minor that I'd like to opt for a simplified sample app. Can you make it a sublayer?